### PR TITLE
Create reusable school info input component

### PR DESCRIPTION
**Pull request by drewsamnick**
This packages up all the pieces needed to collect school info into a single, reusable component. The code is based on [the code from the registration form](https://github.com/code-dot-org/code-dot-org/blob/f8a1084122f17f1c46c902e0e77aa37f8b279972/apps/src/sites/studio/pages/signup.js#L162). I'm going to use this in the new (still work in progress) census teacher banner. Right now I am not going to update the other places that could use this (/yourschool, hoc signup, school info interstitial, registration, user update page.) I suspect that when we go to do that we will find some more tweaks that are needed. So, this is a big step toward a truly reusable component but not the final one. 

A couple of things to note:
* I made it possible to override the field names. This will allow the removal of code like [this](https://github.com/code-dot-org/code-dot-org/blob/f8a1084122f17f1c46c902e0e77aa37f8b279972/apps/src/sites/studio/pages/signup.js#L266) that has to transform to the right names. I'm defaulting to the names required for updating/creating a user.
* I'm omitting the city field in SchoolNotFound. We never actually record the city (school_infos has no column for it) so it doesn't make sense to ask people to fill it in. We do ask for the full_address with a text label that says "City / Town" when useGoogleLocationSearch is set to true. 

Here is an example of what the component looks like: 
screen shot 2017-12-04 at 3 26 48 pm

### CODE REVIEW

File: apps/src/templates/CountryAutocompleteDropdown.jsx [hunk 1]
```diff
@@ -28,9 +28,14 @@ export default class CountryAutocompleteDropdown extends Component {
     showErrorMsg: PropTypes.bool,
     showRequiredIndicator: PropTypes.bool,
     value: PropTypes.string,
+    fieldName: PropTypes.string,
```
Comment (Erin007) [inline apps/src/templates/CountryAutocompleteDropdown.jsx:31]: Should fieldName be required? Same for school dropdown.

Comment (aoby) [reply apps/src/templates/CountryAutocompleteDropdown.jsx:31]: It has a default value below. My impression of this (from the PR description) is that this prop is just for overriding the default when needed.

File: apps/src/templates/SchoolInfoInputs.jsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      schoolName: "user[school_info_attributes][school_name]",
+      schoolState: "user[school_info_attributes][school_state]",
+      schoolZip: "user[school_info_attributes][school_zip]",
+      googleLocation: "user[school_info_attributes][full_address]",
+    },
+  };
+
+  render() {
+    const isUS = this.props.country === 'United States';
+    const outsideUS = !isUS;
```
Comment (Erin007) [inline apps/src/templates/SchoolInfoInputs.jsx:57]: Is defining outsideUs just so the code is more human readable?

File: apps/src/templates/SchoolInfoInputs.jsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      !SCHOOL_TYPES_HAVING_NCES_SEARCH.includes(this.props.schoolType)
+      && this.props.schoolType !== ''
+    );
+    const askForName = SCHOOL_TYPES_HAVING_NAMES.includes(this.props.schoolType);
+    const schoolNameLabel = ['afterschool', 'organization'].includes(this.props.schoolType)
+                          ? i18n.signupFormSchoolOrOrganization()
+                          : i18n.schoolName();
+
+    return (
+      <div style={{width: 600}}>
```
Comment (Erin007) [inline apps/src/templates/SchoolInfoInputs.jsx:69]: This width is wider than some mobile screens; I don't think we want to hardcode it for uses of this component on responsive pages.

File: apps/src/templates/SchoolTypeDropdown.jsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+const singleLineContainerStyles = {
+  display: "table",
+  width: "100%",
+};
+const selectStyle = {
+  width: 390,
+  verticalAlign: 'top',
+  marginBottom: '5px',
+  marginTop: '5px',
+};
```
Comment (Erin007) [inline apps/src/templates/SchoolTypeDropdown.jsx]: More often than not I see styles in one styles object, like [this](https://github.com/code-dot-org/code-dot-org/blob/9873b1c7c25f9037123331ac59f242f1afcca6b4/apps/src/templates/studioHomepages/ResourceCard.jsx#L8).

Comment (aoby) [reply apps/src/templates/SchoolTypeDropdown.jsx]: +1. This is specified in our [styleguide](https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#react)

**Comment by Erin007** [comment]
🎉 For reusability!

File: apps/src/templates/CountryAutocompleteDropdown.jsx (hunk 1 above)
Comment (drewsamnick) [reply apps/src/templates/CountryAutocompleteDropdown.jsx:31]: Yes, this is optional. The default is the value that was previously hard coded, for backward compatibility.

File: apps/src/templates/SchoolInfoInputs.jsx (hunk 2 above)
Comment (drewsamnick) [reply apps/src/templates/SchoolInfoInputs.jsx:57]: Yes, it is for readability. This code was originally from [signup.js](https://github.com/code-dot-org/code-dot-org/blob/f8a1084122f17f1c46c902e0e77aa37f8b279972/apps/src/sites/studio/pages/signup.js#L365) and I kept it as it was.

File: apps/src/templates/SchoolInfoInputs.jsx (hunk 3 above)
Comment (drewsamnick) [reply apps/src/templates/SchoolInfoInputs.jsx:69]: Yeah, I don't love this either. It is effectively what the sign up form does (using a style with a width of 600.)  CountryAutocompleteDropdown also has [a fixed width](https://github.com/code-dot-org/code-dot-org/blob/14fdb203cf7f3284b36e8f0b400b574ca052a13f/apps/src/templates/CountryAutocompleteDropdown.jsx#L12) for part of its layout. Getting it to display well for arbitrary widths was more than I was ready to tackle right now. 

How do you feel about leaving this set to 600 for now? It is one of the things that will need to be updated as this component gets reused in different places.

File: apps/src/templates/SchoolTypeDropdown.jsx (hunk 4 above)
Comment (drewsamnick) [reply apps/src/templates/SchoolTypeDropdown.jsx]: I've seen both. The single object approach seems to be used when there are many styles and the separate ones when there are just a few. I don't have a strong feeling so can change this if you want.

Comment (aoby) [reply apps/src/templates/SchoolTypeDropdown.jsx]: Our styleguide (linked above) explicitly specifies the single object, so I'd prefer that. We _shouldn't_ have the other style, and certainly shouldn't be adding more like that.

> Prefer single object for all styles vs. inlined style objects. Define static styles outside the component, and only dynamic styles in the render method.

Comment (drewsamnick) [reply apps/src/templates/SchoolTypeDropdown.jsx]: Got it. I'll make the change.

**Pull request merged**